scanner.go

```diff
@@ -42,6 +42,26 @@ type Field struct {
 	IsEmbeded bool
 }
 
+type StructInfo struct {
+	Fields []Field
+}
+
+// GetStructInfo will return (and cache) informatoin about the given struct.
+//
+// `targetStruct` should either be a pointer to a struct type, or a
+// reflect.Type object of the structure in question
+func GetStructInfo(targetStruct interface{}) (si StructInfo, err error) {
+	if t, ok := targetStruct.(reflect.Type); ok {
+		if t.Kind() == reflect.Ptr {
+			t = t.Elem()
+		}
+		si.Fields, err = getStructInfoForType(t)
+	} else {
+		_, _, si.Fields, err = getStructInfo(targetStruct)
+	}
+	return
```

Comment:
I don't mind having named return values, they are good for documentation, but I usually prefer to avoid implicit returns, because an implicit return value forces the reader to scroll up to the signature of the function which is a small but unnecessary step to go through. So I usually prefer to do it explicitly even when using named return values:

```suggestion
	return si, err
```